WatchFaceAlphaKotlin/app/src/main/java/com/example/android/wearable/alpha/viewmodels/AnalogWatchFaceViewModel.kt

```diff
@@ -15,123 +15,52 @@
  */
 package com.example.android.wearable.alpha.viewmodels
 
-import androidx.lifecycle.asLiveData
-import androidx.wear.watchface.style.UserStyle
 import com.example.android.wearable.alpha.AnalogWatchFaceService
-import com.example.android.wearable.alpha.data.WatchFaceRepository
-import com.example.android.wearable.alpha.data.db.AnalogWatchFaceEntity
-import com.example.android.wearable.alpha.data.db.WatchFaceArmDimensionsEntity
-import com.example.android.wearable.alpha.utils.COLOR_STYLE_SETTING
-import com.example.android.wearable.alpha.utils.DRAW_HOUR_PIPS_STYLE_SETTING
-import com.example.android.wearable.alpha.utils.WATCH_HAND_LENGTH_STYLE_SETTING
+import com.example.android.wearable.alpha.data.WatchFaceUserSettingChangesDataSource
+import com.example.android.wearable.alpha.data.WatchFaceUserSettingChangesDataSource.WatchFaceUserChanges
 import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.MainScope
 import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.launch
 
 /**
- * View model used for [AnalogWatchFaceService].
+ * View model used for [AnalogWatchFaceService]. Uses flow to receive changes in the data source and
+ * because the renderer is a service without LifeCycle support, handles canceling scope when no
+ * longer needed.
  */
-class AnalogWatchFaceViewModel(private val repository: WatchFaceRepository) {
-
-    // Used to launch coroutines (non-blocking way to insert data).
+@ExperimentalCoroutinesApi
```

Comment:
Nit: I like using the `@OptIn` version for these to clearly show the place where we are choosing to use an experimental piece of the API, which stops the propagating of `@ExperimentalCoroutinesApi` for any class that uses it.